### PR TITLE
DaemonSet Controller and tests to apps/v1

### PR DESCRIPTION
**Pull request by kow3ns**
**What this PR does / why we need it**:
Updates the DaemonSet controller, its integration tests, and its e2e tests to use the apps/v1 API.

**Release note**:
```release-note
The DaemonSet controller, its integration tests, and its e2e tests, have been updated to use the apps/v1 API.
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/extensions.go
Comment (janetkuo) [inline cmd/kube-controller-manager/app/extensions.go]: We should move this into ./apps.go

File: pkg/controller/daemon/daemon_controller.go
Comment (janetkuo) [inline pkg/controller/daemon/daemon_controller.go]: I think it's fine to ignore the returned error here. However, if there's an error, we should set `generation` to `nil` to avoid unexpected behavior.

Comment (janetkuo) [reply pkg/controller/daemon/daemon_controller.go]: Do we need to print a warning here if there's an error?

File: pkg/controller/daemon/daemon_controller.go
Comment (janetkuo) [inline pkg/controller/daemon/daemon_controller.go]: ditto

File: pkg/controller/daemon/update.go
Comment (janetkuo) [inline pkg/controller/daemon/update.go]: Here too

File: pkg/controller/daemon/util/daemonset_util.go
Comment (janetkuo) [inline pkg/controller/daemon/util/daemonset_util.go]: Need to copy labels from old template to new template

File: pkg/controller/daemon/util/daemonset_util_test.go
Comment (janetkuo) [inline pkg/controller/daemon/util/daemonset_util_test.go]: This is used several times in this file. Worth adding a `func(int) *int64`

File: test/e2e/apps/daemon_set.go
Comment (janetkuo) [inline test/e2e/apps/daemon_set.go]: This tests adoption using hash (should work even when templateGeneration doesn't match). We should keep this and remove parts about templateGeneration. 

Also change the description to "Should adopt existing pods when creating a RollingUpdate DaemonSet".

File: test/e2e/apps/daemon_set.go
Comment (janetkuo) [inline test/e2e/apps/daemon_set.go]: Remove setting templateGeneration and all checks for it

File: test/e2e/apps/daemon_set.go
Comment (janetkuo) [inline test/e2e/apps/daemon_set.go]: Remove `templateGeneration` from the func

File: test/e2e/apps/daemon_set.go
Comment (janetkuo) [inline test/e2e/apps/daemon_set.go]: Step 3 and 4 can be removed. Need to update comments of step 5 (becomes step 3).

File: test/e2e/apps/daemon_set.go
Comment (janetkuo) [inline test/e2e/apps/daemon_set.go]: Step 6 and 7 can be removed.

File: pkg/controller/daemon/util/daemonset_util.go
Comment (janetkuo) [reply pkg/controller/daemon/util/daemonset_util.go]: You're right, I didn't notice that newTemplate is already a copy of template.

File: test/e2e/apps/daemon_set.go
Comment (kow3ns) [reply test/e2e/apps/daemon_set.go]: I think we should do this as an integration test. I can see the value for e2e when TemplateGeneration was a supported, user facing feature. The template hash, and its interaction with orphaning and adoption, is an implementation detail of the controller that requires only the API server and the controller manager to test.

Comment (janetkuo) [reply test/e2e/apps/daemon_set.go]: Moving it to integration sounds good.

**Review by janetkuo — APPROVED** [review]

**Review by dhilipkumars — COMMENTED** [review]
Thanks, now linked to the umbrella issue.  Could you squash all the commits please.
/lgtm

**Comment by janetkuo** [comment]
/assign @deads2k 

PTAL

**Comment by kow3ns** [comment]
@dhilipkumars The commits are purposefully left separate so that its easy to see what was changed in the controllers, in the client, and in the tests.

File: pkg/controller/daemon/daemon_controller.go
Comment (tnozicka) [inline pkg/controller/daemon/daemon_controller.go]: if we are ignoring the error, could we at least log it?

File: pkg/controller/daemon/daemon_controller.go
Comment (tnozicka) [inline pkg/controller/daemon/daemon_controller.go]: log

File: pkg/controller/daemon/update.go
Comment (tnozicka) [inline pkg/controller/daemon/update.go]: log

File: pkg/controller/daemon/daemon_controller.go
Comment (kow3ns) [reply pkg/controller/daemon/daemon_controller.go]: Janet and I discussed this, and we have a concern about useless noise in the logs

File: test/e2e/apps/daemon_set.go
Comment (crimsonfaith91) [reply test/e2e/apps/daemon_set.go]: I came out with the PR to make it an integration test: #59013
I will wait until this PR is merged, change my test to use `apps/v1`, and then ask for review.

File: pkg/controller/daemon/daemon_controller.go
Comment (tnozicka) [reply pkg/controller/daemon/daemon_controller.go]: if the concern is noise, loglevels are the distinction

**Pull request merged**
